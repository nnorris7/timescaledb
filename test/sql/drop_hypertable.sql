\ir include/create_single_db.sql

SELECT * from _timescaledb_catalog.hypertable;
SELECT * from _timescaledb_catalog.dimension;

CREATE TABLE should_drop (time timestamp, temp float8);
SELECT create_hypertable('should_drop', 'time');

-- Calling create hypertable again will increment hypertable ID
-- although no new hypertable is created. Make sure we can handle this.
SELECT create_hypertable('should_drop', 'time', if_not_exists => true);
SELECT * from _timescaledb_catalog.hypertable;
SELECT * from _timescaledb_catalog.dimension;
DROP TABLE should_drop;

CREATE TABLE should_drop (time timestamp, temp float8);
SELECT create_hypertable('should_drop', 'time');

INSERT INTO should_drop VALUES (now(), 1.0);
SELECT * from _timescaledb_catalog.hypertable;
SELECT * from _timescaledb_catalog.dimension;

