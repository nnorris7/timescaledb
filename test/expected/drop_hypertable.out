\ir include/create_single_db.sql
SET client_min_messages = WARNING;
DROP DATABASE IF EXISTS single;
SET client_min_messages = NOTICE;
CREATE DATABASE single;
\c single
CREATE EXTENSION IF NOT EXISTS timescaledb;
SELECT * from _timescaledb_catalog.hypertable;
 id | schema_name | table_name | associated_schema_name | associated_table_prefix | num_dimensions 
----+-------------+------------+------------------------+-------------------------+----------------
(0 rows)

SELECT * from _timescaledb_catalog.dimension;
 id | hypertable_id | column_name | column_type | num_slices | partitioning_func_schema | partitioning_func | interval_length 
----+---------------+-------------+-------------+------------+--------------------------+-------------------+-----------------
(0 rows)

CREATE TABLE should_drop (time timestamp, temp float8);
SELECT create_hypertable('should_drop', 'time');
 create_hypertable 
-------------------
 
(1 row)

-- Calling create hypertable again will increment hypertable ID
-- although no new hypertable is created. Make sure we can handle this.
SELECT create_hypertable('should_drop', 'time', if_not_exists => true);
NOTICE:  hypertable should_drop already exists, skipping
 create_hypertable 
-------------------
 
(1 row)

SELECT * from _timescaledb_catalog.hypertable;
 id | schema_name | table_name  | associated_schema_name | associated_table_prefix | num_dimensions 
----+-------------+-------------+------------------------+-------------------------+----------------
  1 | public      | should_drop | _timescaledb_internal  | _hyper_1                |              1
(1 row)

SELECT * from _timescaledb_catalog.dimension;
 id | hypertable_id | column_name |         column_type         | num_slices | partitioning_func_schema | partitioning_func | interval_length 
----+---------------+-------------+-----------------------------+------------+--------------------------+-------------------+-----------------
  1 |             1 | time        | timestamp without time zone |            |                          |                   |   2592000000000
(1 row)

DROP TABLE should_drop;
CREATE TABLE should_drop (time timestamp, temp float8);
SELECT create_hypertable('should_drop', 'time');
 create_hypertable 
-------------------
 
(1 row)

INSERT INTO should_drop VALUES (now(), 1.0);
SELECT * from _timescaledb_catalog.hypertable;
 id | schema_name | table_name  | associated_schema_name | associated_table_prefix | num_dimensions 
----+-------------+-------------+------------------------+-------------------------+----------------
  2 | public      | should_drop | _timescaledb_internal  | _hyper_2                |              1
(1 row)

SELECT * from _timescaledb_catalog.dimension;
 id | hypertable_id | column_name |         column_type         | num_slices | partitioning_func_schema | partitioning_func | interval_length 
----+---------------+-------------+-----------------------------+------------+--------------------------+-------------------+-----------------
  2 |             2 | time        | timestamp without time zone |            |                          |                   |   2592000000000
(1 row)

